src/App.js

```diff
@@ -17,21 +18,33 @@ const makeRequest = async (endpoint, data) => {
 
 export function App() {
   const [synthetix, updateSynthetix] = useSynthetix();
-  const { walletAddress, connect, signature, signer } = synthetix;
+  const { walletAddress, connect, tokens, signer } = synthetix;
 
   const signupMutation = useMutation({
-    mutationFn: (data) => makeRequest('signup', data),
+    mutationFn: (data) => makeUnauthenticatedRequest('signup', data),
     onSuccess: ({ nonce }) =>
       signer.signMessage(nonce).then((signedMessage) => {
-        verificationMutation.mutate({ signedMessage });
+        verificationMutation.mutate({ nonce, signedMessage });
       }),
   });
 
   const verificationMutation = useMutation({
-    mutationFn: (data) => makeRequest('verify', data),
-    onSuccess: ({ signature }) => updateSynthetix({ signature }),
+    mutationFn: (data) => makeUnauthenticatedRequest('verify', data),
+    onSuccess: ({ token }) => {
+      const tokensFromLocalStorage = getTokensFromLocalStorage();
+      tokensFromLocalStorage[walletAddress] = token;
+      window.localStorage.setItem('tokens', JSON.stringify(tokensFromLocalStorage));
+      updateSynthetix({ tokens: tokensFromLocalStorage });
+    },
   });
 
+  const logout = () => {
+    const tokens = getTokensFromLocalStorage();
+    delete tokens[walletAddress];
+    window.localStorage.setItem('tokens', JSON.stringify(tokens));
```

Comment:
```suggestion
    window.localStorage.clear();
```

Reply:
we keep things in object to avoid accidental collisions, but we dont need to store multiple tokens forever. When you logout - clear all things. When you change wallet - yeah that's ok, just change token without logout.